src/snowflake/snowpark/dataframe.py

```diff
@@ -1644,6 +1621,95 @@ def unpivot(
             )
         return self._with_plan(unpivot_plan)
 
+    @df_api_usage
+    def exclude_columns(
+        self,
+        *cols: Union[ColumnOrName, Iterable[ColumnOrName]],
+    ) -> "DataFrame":
+        """Excludes a list of columns from the table. In a query, it is specified in the EXCLUDE clause.
+
+        This is functionally equivalent to calling :func:`drop()`. Unlike drop(), exclude() executes lazily using the SELECT * EXCLUDE clause.
+        Note that unlike :func:`drop()`, excluding a column that does not exist will not result in an exception immediately.
+
+        Example::
+
+            >>> df = session.create_dataframe([
+            ...     (1, 'electronics', "JAN", 100),
+            ...     (2, 'clothes', "FEB", 200)
+            ... ], schema=["empid", "dept", "month", "sales"])
+            >>> df = df.exclude_columns("month")
+            >>> df.show()
+            -----------------------------------
+            |"EMPID"  |"DEPT"       |"SALES"  |
+            -----------------------------------
+            |1        |electronics  |100      |
+            |2        |clothes      |200      |
+            -----------------------------------
+            <BLANKLINE>
+
+        Args:
+            *cols: The columns to exclude as :class:`str`, :class:`Column` or a list of those.
+        """
+        if not cols:
+            raise ValueError("The input of exclude() cannot be empty")
+        exprs = parse_positional_args_to_list(*cols)
+        names = self._get_column_names_from_column_or_name_list(exprs)
+        normalized_names = {quote_name(n) for n in names}
+        exclude_plan = Exclude(normalized_names, self._plan)
+
+        if self._select_statement:
+            return self._with_plan(
+                SelectStatement(
+                    from_=SelectSnowflakePlan(
+                        exclude_plan, analyzer=self._session._analyzer
+                    ),
+                    analyzer=self._session._analyzer,
+                )
+            )
+        return self._with_plan(exclude_plan)
```

Comment:
Might have to add tests in simplifier suite to verify that this works. My guess is it should

Reply:
Sure; I think a clean approach would be to add a second PR for simplifier changes (optimization and test cases)